### ISSUE TITLE
Running Different Application Frameworks with uv Command

### ISSUE THREAD

**Issue by HeangSok-2** [astral-sh/uv#9777]
At the moment, when we run our application using the uv command, for example:
````bash
uv run --package app1 apps/app1/main.py
````
The question arises: What if app1 utilizes Google Cloud Functions Framework or FastAPI? Can we adapt the command to handle these frameworks? For instance:

1. For Google Cloud Functions Framework:
````bash
uv run --package app1 "functions-framework --source apps/app1/main.py --target main --port 8080"
```` 
2. For FastAPI:
````bash
uv run --package app1 "uvicorn apps.app1.main:app --workers 1 --host 0.0.0.0 --port 8000 --reload"
````

This approach could streamline development while accommodating various frameworks in a separate sandbox. Is this feasible?

**Comment by zanieb** [comment]
Yeah, use `dependency-groups` or `extras` or `--with <pkg>` for an ad-hoc dependency

https://docs.astral.sh/uv/concepts/projects/dependencies/
